.github/workflows/publish.yaml

```diff
@@ -20,13 +20,15 @@ jobs:
           version: latest
 
       - name: Install Node.js
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
```

Comment:
let's use node 22 instead of 18

and also
```
- uses: actions/checkout@v3
```
```
      - name: Install pnpm
        uses: pnpm/action-setup@v4
        with:
          version: 9
```

across all other workflows